pyseir/models/demographics.py

```diff
@@ -0,0 +1,247 @@
+        rate = Demographics.MEDIAN_RATE_CONSTANT * (rt - 1.0) / self.dwell_time if rt > 1.0 else 0.0
+
+        new_median = (self.median_age + rate * self.driver_age + default_age / self.dwell_time) / (
+            1.0 + rate + 1.0 / self.dwell_time
+        )
+
+        self.median_age = new_median
+
+    def update_by_contacting_another(self, contacts_type, adding, from_demo, dt=1.0):
+        """
+        Update demographics based on new set of individuals being added from a source based on contacts
+        """
+        raw_new = Demographics.CONTACTS[contacts_type] @ self.as_array() * from_demo.as_array()
+        scaled_new = raw_new * (adding / raw_new.sum())
+        new_counts = self.count * self.as_array() * (1.0 - dt / self.dwell_time) + scaled_new
+
+        new_fractions = new_counts / new_counts.sum()
+        self.from_array(new_fractions)
+
+    @staticmethod
+    def infer_median_age_function(t_list, rt_f):
+        t_list = extend_tlist_at_beginning(t_list, 30)
+        default_usa_f = Demographics.median_age_f()
+        demo = Demographics(median_age=default_usa_f(t_list[0]))
+
+        values = {t_list[0]: demo.get_median_age()}
+        for t in t_list:
+            if t < 135.0:  # don't use this early in the pandemic
+                demo = Demographics(median_age=default_usa_f(t))
+            else:
+                demo.evolve_median(rt_f(t), default_usa_f(t))
+            values[t] = demo.get_median_age()
+
+        def ftn(t):
+            if t <= t_list[0]:
+                return values[t_list[0]]
+            elif t >= t_list[-1]:
+                return values[t_list[-1]]
+            else:
+                return values[t]
+
+        return ftn
+
+    @staticmethod
+    def default():
+        r_lo_mid = 1.1
+        r_hi_mid = 0.4
+        mid = 1.0 / (1.0 + r_lo_mid + r_hi_mid)
+        return [r_lo_mid * mid, mid, r_hi_mid * mid]
+
+    @staticmethod
+    def age_fractions_from_median(median_age):
+        center_pref = 4.0
+        if median_age < 35.0 or median_age > 65.0:
+            assert False
+        k = (median_age - 35.0) / 30.0
+
+        # Coefficients of quadratic equation for m - fraction of medium age
+        a = 2.0 / center_pref + 2 * k * (1.0 - k)
+        b = 1.0
+        c = -0.5
+
+        # Solution of quadratic equation
+        m = (-b + math.sqrt(b ** 2 - 4.0 * a * c)) / (2.0 * a)
+
+        # Then solve for young and old
+        y = 0.5 - k * m
+        o = 1.0 - m - y
+
+        avg_vs_median = (17.5 * y + 50.0 * m + 82.5 * o) - median_age
+
+        return (y, m, o)
+
+    @staticmethod
+    def median_age_f(state=None):
+        """
+        Median age of covid cases for all of US and specific individual states
+        """
+        if state not in ["FL"]:  # median age for country as a whole
+            # From https://www.cdc.gov/coronavirus/2019-ncov/covid-data/covidview/index.html
+            data = {
+                68: 56.3,
```

Comment:
Just flagging here that we may want to update these values as we discussed.

Reply:
Brett has found a bunch more data on the CDC site so I will generalize this better once that analysis is done - but that will be after this PR. Need this merged so integration work can proceed - this code is not yet used in production.